### ISSUE TITLE
bpf tc can't load by tc or cargo bpf load

### ISSUE THREAD

**Issue by Sherlock-Holo** [foniod/redbpf#204]
I'm using Archlinux

> Linux sherlockholo-pc 5.14.14-arch1-1 #1 SMP PREEMPT Wed, 20 Oct 2021 21:35:18 +0000 x86_64 GNU/Linux

I try this codes
```rust
#![no_std]
#![no_main]

use redbpf_probes::tc::prelude::*;
// use redbpf_probes::xdp::prelude::*;

program!(0xFFFFFFFE, "GPL");

#[tc_action]
fn test(skb: SkBuff) -> TcActionResult {
    Ok(TcAction::Ok)
}
```

and build with `cargo bpf build`

When I try to load it by `sudo tc filter add ingress bpf da obj target/bpf/programs/learn_bpf/learn_bpf.elf sec tc_action/test`

it reports

> Error fetching program/map!
> Unable to load program

also I try to use cargo-bpf, `sudo -E cargo bpf load -i lo target/bpf/programs/learn_bpf/learn_bpf.elf`, it just stuck without any output.

How should I do to load the bpf tc program?

**Comment by d0u9** [comment]
Is the rust code for bpf are directly compiled by llvm? Something like:

```
llvm [options] -arch bpf -o bpf.rust
```

Or multiple steps are utilized to translate .rust to .xxx to .bpf?

**Comment by d0u9** [comment]
I just found that article, and that is much complex than I thought before.

**Comment by rhdxmr** [comment]
@Sherlock-Holo 

> @rhdxmr well... that's so hard to try to remove the `.text` section :(
> 
> I have try similar bpf tc codes, written by c, and compile by `clang -I ./headers -Wall -O3 -target bpf -c main.c -o main.o` (please ignore the -I ./headers, actually there's nothing there)
> 
> and load by tc success directly, no need to remove any section
> 
> is a plan to let redbpf can load by tc directly without removing some section?

I agreed with your opinion so I tried to remove `.text` section from ELF relocatable.
The problem has been solved by this PR #224
The fix will be released soon with a RedBPF `v2.2.0` release.
Hence, with the `v2.2.0` perhaps don't need to struggle with removing the `.text` section manually.

Thanks,
